ios/browser/api/sync/brave_sync_worker.cc

```diff
@@ -326,6 +337,30 @@ void BraveSyncWorker::DeleteDevice(const std::string& device_guid) {
   brave_sync::DeleteDevice(sync_service, device_info_service, device_guid);
 }
 
+void BraveSyncWorker::SetJoinSyncChainCallback(
+    base::OnceCallback<void(const bool&)> callback) {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  syncer::BraveSyncServiceImpl* sync_service = GetSyncService();
+
+  if (!sync_service) {
+    return;
+  }
+
+  sync_service->SetJoinChainResultCallback(base::BindOnce(std::move(callback)));
```

Comment:
I don't think you have to re-bind this, no? i.e. `sync_service->SetJoinChainResultCallback(std::move(callback))`

Reply:
Fixed at https://github.com/brave/brave-core/pull/17369/commits/6c7ea395fef0dc1a7f0bcbbc849a0367183aeee7